lib/man_task.rb

```diff
@@ -16,12 +16,57 @@
 # you may find current contact information at www.suse.com
 
 class ManTask
-  def man
+  def self.compile_documentation
+    docs = Inspector.all_scopes.map do |scope|
+      scope_doc = "* #{scope}\n\n"
+      scope_doc += YAML.load_file(
+        File.join(Machinery::ROOT, "plugins/#{scope}/#{scope}.yml")
+      )[:description]
+      scope_doc + "\n"
+    end.join
+    File.write("manual/docs/machinery_main_scopes.1.md", docs)
+    Dir.chdir(File.join(Machinery::ROOT, "manual")) do
+      Cheetah.run("mkdocs", "build")
+    end
+  end
+
+  def man(options)
+    if options[:html]
+      man_html(options)
+    else
+      man_system
+    end
+  end
+
+  def man_system
     LocalSystem.validate_existence_of_package("man")
-    system("man", man_path)
+    system("man", File.join(Machinery::ROOT, "man/generated/machinery.1.gz"))
   end
 
-  def man_path
-    File.join(Machinery::ROOT, "man/generated/machinery.1.gz")
+  def man_html(options)
+    unless File.exist?(File.join(Machinery::ROOT, "manual/site"))
+      Machinery::Ui.warn(
+        "The documentation was not generated yet. Please make sure that 'mkdocs' is installed on" \
+        "your system and run 'rake man_pages:compile_documentation' from the machinery directory"
```

Comment:
@mauromorales at least I do it always with backticks. I "grepped" through the `lib/` directory and it seems that we use it more often for commands.